jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/ResponseHttpFields.java

```diff
@@ -113,15 +129,19 @@ public boolean hasNext()
             @Override
             public HttpField next()
             {
-                return i.next();
+                _current = i.next();
+                return _current;
             }
 
             @Override
             public void remove()
             {
                 if (_committed.get())
                     throw new UnsupportedOperationException("Read Only");
+                if (isPersistent(_current))
+                    throw new IllegalStateException("Persistent field");
                 i.remove();
+                _current = null;
```

Comment:
You should also throw ISE when `_current` is null according to the `Iterator.remove()` javadoc.

Reply:
The i.remove() on the previous line will do that.  But I will add an explicit check